pkg/graph/schema.graphql

```diff
@@ -641,62 +642,221 @@ https://gqlgen.com/reference/changesets/
+  communicationNote:     String
+  participantAssumeRisk: Boolean
   riskType:              ParticipantRiskType
-  riskOther:             String             
-  riskNote:              String             
-  willRiskChange:        Boolean             
-  willRiskChangeNote:    String             
+  riskOther:             String
+  riskNote:              String
+  willRiskChange:        Boolean
+  willRiskChangeNote:    String
 
   #Page 4
-  coordinateWork:          Boolean       
-  coordinateWorkNote:      String       
-  gainsharePayments:       Boolean       
-  gainsharePaymentsTrack: Boolean       
-  gainsharePaymentsNote:   String       
+  coordinateWork:          Boolean
+  coordinateWorkNote:      String
+  gainsharePayments:       Boolean
+  gainsharePaymentsTrack: Boolean
+  gainsharePaymentsNote:   String
   participantsIds:         [ParticipantsIDType!]
-  participantsIdsOther:    String       
-  participantsIDSNote:     String       
+  participantsIdsOther:    String
+  participantsIDSNote:     String
 
   #Page 5
   providerAdditionFrequency:      FrequencyType
-  providerAdditionFrequencyOther: String       
-  providerAdditionFrequencyNote:  String       
+  providerAdditionFrequencyOther: String
+  providerAdditionFrequencyNote:  String
   providerAddMethod:              [ProviderAddType!]
-  providerAddMethodOther:         String       
-  providerAddMethodNote:          String       
+  providerAddMethodOther:         String
+  providerAddMethodNote:          String
   providerLeaveMethod:            [ProviderLeaveType!]
-  providerLeaveMethodOther:       String       
-  providerLeaveMethodNote:        String       
-  providerOverlap:                OverlapType  
-  providerOverlapHierarchy:       String       
-  providerOverlapNote:            String     
+  providerLeaveMethodOther:       String
+  providerLeaveMethodNote:        String
+  providerOverlap:                OverlapType
+  providerOverlapHierarchy:       String
+  providerOverlapNote:            String
 
 }
 
+"""
+PlanPayments is the task list section that deals with information regarding Payments
+"""
+type PlanPayments {
+  id: UUID!
+  modelPlanID: UUID!
+
+  # Page 1
+  fundingSource:                      [FundingSource!]
+  fundingSourceTrustFund:             String
+  fundingSourceNote:                  String
+  fundingSourceR:                     [FundingSource!]
+  fundingSourceRTrustFund:            String
+  fundingSourceROther:                String
+  fundingSourceRNote:                 String
+  payRecipients:                      [PayRecipient!]
+  payRecipientsOtherSpecification:    String
+  payRecipientsNote:                  String
+  payType:                            [PayType!]
+  payTypeNote:                        String
+
+  # Page 2
+  payClaims:                                      [ClaimsBasedPayType!]
+  payClaimsOther:                                 String
+  shouldAnyProvidersExcludedFFSSystems:           Boolean!
```

Comment:
@OddTomBrooks , the query to the backend fails. It looks like it is because the GQL defines some of these Boolean and Time columns as not null eg (Boolean! Time!). These should all allow null, as depending on where we are in the lifecycle of the form, the answers will be null.

Here is the query I used to test this. I'd recommend using this in Postman to verify that the code changes handle all the fields that should be nullable

```gql
mutation updatePlanPayments
{
    updatePlanPayments (id: "a18d2f8c-e5cb-4c5d-b21c-3554dbec9c51",changes:     
    {
        # # Page 1
        # fundingSource:
        # fundingSourceTrustFund:
        fundingSourceNote: "ello"
        # fundingSourceR:
        # fundingSourceRTrustFund:
        # fundingSourceROther:
        # fundingSourceRNote:
        # payRecipients:
        # payRecipientsOtherSpecification:
        # payRecipientsNote:
        # payType:
        # payTypeNote:

        # # Page 2
        # payClaims:
        # payClaimsOther:
        # shouldAnyProvidersExcludedFFSSystems:
        # shouldAnyProviderExcludedFFSSystemsNote:
        # changesMedicarePhysicianFeeSchedule:
        # changesMedicarePhysicianFeeScheduleNote:
        # affectsMedicareSecondaryPayerClaims:
        # affectsMedicareSecondaryPayerClaimsHow:
        # affectsMedicareSecondaryPayerClaimsNote:
        # payModelDifferentiation:

        # # Page 3
        # creatingDependenciesBetweenServices:
        # creatingDependenciesBetweenServicesNote:
        # needsClaimsDataCollection:
        # needsClaimsDataCollectionNote:
        # providingThirdPartyFile:
        # isContractorAwareTestDataRequirements:

        # # Page 4
        # beneficiaryCostSharingLevelAndHandling:
        # waiveBeneficiaryCostSharingForAnyServices:
        # waiveBeneficiaryCostSharingServiceSpecification:
        # waiverOnlyAppliesPartOfPayment:
        # waiveBeneficiaryCostSharingNote:

        # # Page 5
        # nonClaimsPayments:
        # nonClaimsPaymentOther:
        # paymentCalculationOwner:
        # numberPaymentsPerPayCycle:
        # numberPaymentsPerPayCycleNotes:
        # sharedSystemsInvolvedAdditionalClaimPayment:
        # sharedSystemsInvolvedAdditionalClaimPaymentNote:
        # planningToUseInnovationPaymentContractor:
        # planningToUseInnovationPaymentContractorNote:
        # fundingStructure:

        # # Page 6
        # expectedCalculationComplexityLevel:
        # expectedCalculationComplexityLevelNote:
        # canParticipantsSelectBetweenPaymentMechanisms:
        # canParticipantsSelectBetweenPaymentMechanismsHow:
        # canParticipantsSelectBetweenPaymentMechanismsNote:
        # anticipatedPaymentFrequency:
        # anticipatedPaymentFrequencyOther:
        # anticipatedPaymentFrequencyNotes:

        # # Page 7
        # willRecoverPayments:
        # willRecoverPaymentsNotes:
        # anticipateReconcilingPaymentsRetrospectively:
        # anticipateReconcilingPaymentsRetrospectivelyNotes:
        # paymentStartDate:
        # paymentStartDateNotes:

        # # Meta
        # createdBy:
        # createdDts:
        # modifiedBy:
        # modifiedDts:
        # status:


    }) {
  
    id
    modelPlanID

    # Page 1
    fundingSource
    fundingSourceTrustFund
    fundingSourceNote
    fundingSourceR
    fundingSourceRTrustFund
    fundingSourceROther
    fundingSourceRNote
    payRecipients
    payRecipientsOtherSpecification
    payRecipientsNote
    payType
    payTypeNote

    # Page 2
    payClaims
    payClaimsOther
    shouldAnyProvidersExcludedFFSSystems
    shouldAnyProviderExcludedFFSSystemsNote
    changesMedicarePhysicianFeeSchedule
    changesMedicarePhysicianFeeScheduleNote
    affectsMedicareSecondaryPayerClaims
    affectsMedicareSecondaryPayerClaimsHow
    affectsMedicareSecondaryPayerClaimsNote
    payModelDifferentiation

    # Page 3
    creatingDependenciesBetweenServices
    creatingDependenciesBetweenServicesNote
    needsClaimsDataCollection
    needsClaimsDataCollectionNote
    providingThirdPartyFile
    isContractorAwareTestDataRequirements

    # Page 4
    beneficiaryCostSharingLevelAndHandling
    waiveBeneficiaryCostSharingForAnyServices
    waiveBeneficiaryCostSharingServiceSpecification
    waiverOnlyAppliesPartOfPayment
    waiveBeneficiaryCostSharingNote

    # Page 5
    nonClaimsPayments
    nonClaimsPaymentOther
    paymentCalculationOwner
    numberPaymentsPerPayCycle
    numberPaymentsPerPayCycleNotes
    sharedSystemsInvolvedAdditionalClaimPayment
    sharedSystemsInvolvedAdditionalClaimPaymentNote
    planningToUseInnovationPaymentContractor
    planningToUseInnovationPaymentContractorNote
    fundingStructure

    # Page 6
    expectedCalculationComplexityLevel
    expectedCalculationComplexityLevelNote
    canParticipantsSelectBetweenPaymentMechanisms
    canParticipantsSelectBetweenPaymentMechanismsHow
    canParticipantsSelectBetweenPaymentMechanismsNote
    anticipatedPaymentFrequency
    anticipatedPaymentFrequencyOther
    anticipatedPaymentFrequencyNotes

    # Page 7
    willRecoverPayments
    willRecoverPaymentsNotes
    anticipateReconcilingPaymentsRetrospectively
    anticipateReconcilingPaymentsRetrospectivelyNotes
    paymentStartDate
    paymentStartDateNotes

    # Meta
    createdBy
    createdDts
    modifiedBy
    modifiedDts
    status

    }
}
``` 

![image](https://user-images.githubusercontent.com/99294987/178291695-d87600e8-b549-4107-9ad0-c7c2cd581572.png)

List of the error response
```gql
{
    "errors": [
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "shouldAnyProvidersExcludedFFSSystems"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "changesMedicarePhysicianFeeSchedule"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "affectsMedicareSecondaryPayerClaims"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "creatingDependenciesBetweenServices"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "providingThirdPartyFile"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "isContractorAwareTestDataRequirements"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "waiveBeneficiaryCostSharingForAnyServices"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "waiverOnlyAppliesPartOfPayment"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "expectedCalculationComplexityLevel"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "canParticipantsSelectBetweenPaymentMechanisms"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "willRecoverPayments"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "anticipateReconcilingPaymentsRetrospectively"
            ]
        },
        {
            "message": "the requested element is null which the schema does not allow",
            "path": [
                "updatePlanPayments",
                "paymentStartDate"
            ]
        }
    ],
    "data": null
}
```